### PR TITLE
SDIT-1334 Add a test app to test consuming the library

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/testapp/TestApp.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.justice.digital.hmpps.testapp
+
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.runApplication
+
+@SpringBootApplication
+class TestApp
```
Comment (mikehalmamoj) [inline test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/testapp/TestApp.kt:7]: Added a module with a copy of the latest hmpps-template-kotlin so we can write integration where we consume the library.

File: test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/testapp/resource/TestAppResource.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.web.bind.annotation.RestController
+import java.time.LocalDateTime
+
+@RestController
+class TestAppResource {
+
+  @PreAuthorize("hasRole('ROLE_TEST_APP')")
+  @RequestMapping("/time")
+  fun getTime() = LocalDateTime.now()
+}
```
Comment (mikehalmamoj) [inline test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/testapp/resource/TestAppResource.kt:14]: The only addition to the test app over the template is a secured endpoint.

File: test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/testapp/integration/resource/TestAppResourceIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.testapp.integration.IntegrationTestBase
+import java.time.LocalDate
+
+class TestAppResourceIntegrationTest : IntegrationTestBase() {
+
+  @Nested
+  inner class TimeEndpoint {
```
Comment (mikehalmamoj) [inline test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/testapp/integration/resource/TestAppResourceIntegrationTest.kt:12]: And an integration test to test the additional secured endpoint. Note that we're not consuming any features from the new library yet, this is a precursor to adding them.

File: hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/HmppsKotlinProperties.kt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package uk.justice.gov.hmpps.kotlin
+package uk.gov.justice.hmpps.kotlin
```
Comment (mikehalmamoj) [inline hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/HmppsKotlinProperties.kt:1]: Just corrected the package names in the library - it still does nothing (yet).

File: dps-gradle-spring-boot-suppressions.xml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (petergphillips) [inline dps-gradle-spring-boot-suppressions.xml:1]: can't really remember - but don't think this file needs to be checked in?  IIRC the trivy one does but not this one...

File: test-app/.circleci/config.yml-z [hunk 6]
```diff
@@ -0,0 +1,144 @@
+version: 2.1
```
Comment (petergphillips) [inline test-app/.circleci/config.yml-z:1]: yml-z file?

File: test-app/.github/workflows/rename_template_project_pr.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
+name: rename-project-create-pr
```
Comment (petergphillips) [inline test-app/.github/workflows/rename_template_project_pr.yml:1]: do you need this for the test app?

Comment (mikehalmamoj) [reply test-app/.github/workflows/rename_template_project_pr.yml:1]: As discussed I'm just pulling everything in from the template to make it easier to see how the template has drifted in future.

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**